vault/auth.go

```diff
@@ -148,6 +155,63 @@ func (v *Vault) UserPassLogin() string {
 	return secret.Auth.ClientToken
 }
 
+// EC2Login - AWS EC2 auth backend
+func (v *Vault) EC2Login() string {
+	if env.Getenv("VAULT_AUTH_AWS_METHOD", "ec2") != "ec2" {
+		return ""
+	}
+
+	if skip := env.Getenv("VAULT_AUTH_AWS_EC2_SKIP", ""); skip != "" {
```

Comment:
As far as moving it last in order, that's _probably_ fine. There are cases where I use gomplate in CI environments and pass it a `VAULT_TOKEN` var so it can auth correctly. Those CI environments are generally in EC2, so it'd be good to not get tripped up there.